src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machineList;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machineList = new BulldozerProducer().get();
+                break;
+
+            case "Excavator":
+                machineList = new ExcavatorProducer().get();
+                break;
+
+            case "Truck":
+                machineList = new TruckProducer().get();
+                break;
+
+            default:
+                machineList = new ArrayList<>();
+        }
+        return new ArrayList<>(machineList);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. The list `machineList` is already local, but you can directly return the result of the `get()` method inside the switch-case blocks to avoid unnecessary variable assignment.